framework/src/modules/interoperability/mainchain/store.ts

```diff
@@ -12,9 +12,139 @@
  * Removal or modification of this copyright notice is prohibited.
  */
 
-//  TODO: extends BaseInteroperabilityStore
-// eslint-disable-next-line  @typescript-eslint/no-extraneous-class
-export class MainchainInteroperabilityStore {
-	// public async forward(ccu: CCUpdateParams, ccm: CCMsg): Promise<void> {};
-	// public async bounce(ccu: CCUpdateParams, ccm: CCMsg): Promise<void> {};
+/*
+ * Copyright © 2022 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+import { BaseInteroperabilityStore } from '../base_interoperability_store';
+import {
+	LIVENESS_LIMIT,
+	MAINCHAIN_ID,
+	MODULE_ID_INTEROPERABILITY,
+	STORE_PREFIX_CHAIN_DATA,
+	STORE_PREFIX_TERMINATED_STATE,
+} from '../constants';
+import {
+	CCMsg,
+	CCUpdateParams,
+	ChainAccount,
+	SendInternalContext,
+	TerminatedStateAccount,
+} from '../types';
+import { chainAccountSchema, terminatedChain } from '../schema';
+
+export class MainchainInteroperabilityStore extends BaseInteroperabilityStore {
+	public async isLive(chainID: number, timestamp: number): Promise<boolean> {
+		if (chainID === MAINCHAIN_ID) {
```

Comment:
i think code referenced is the old one
https://github.com/LiskHQ/lips/pull/131/files#diff-a3ced5b919eed85370156537064b46119c1e19204e1d348b91f757a0f5ca8d8bR756-R768
Here is the updated one on PR.

(from the above link, the mainchain check is not required since this class is for mainchain)

Reply:
In the updated issue it just says "Implement isLive function definition as defined in LIP 0045" without any reference or links. So naturally I went by the logic mentioned in [LIP 0045](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0045.md). Anyways I'll update as per the PR.